salt/hosts/init.sls

```diff
@@ -1,28 +1,13 @@
-{%- set minealias    = salt['pillar.get']('hostsfile:alias', 'network.ip_addrs') %}
-{%- set minions      = salt['pillar.get']('hostsfile:minions', '*') %}
-{%- set pillar_hosts = salt['pillar.get']('hostsfile:hosts', {}) %}
-{%- set mine_hosts   = salt['mine.get'](minions, minealias) %}
-
-{%- set hosts = {} %}
-{%- if mine_hosts is defined %}
-{%-   do hosts.update(mine_hosts) %}
-{%- endif %}
-{%- do hosts.update(pillar_hosts) %}
-
-{%- for fqdn, addrlist in hosts.items() %}
-{{ fqdn }}-host-entry:
+# Internally we refer to hosts by their machine ID, which is also used as the minion ID.
+{%- for minion_id, addrlist in salt['mine.get']('*', 'network.ip_addrs').items() %}
+{{ minion_id }}-host-entry:
   host.present:
 {% if addrlist is string %}
     - ip: {{ addrlist }}
 {% else %}
     - ip: {{ addrlist|first }}
 {% endif %}
```

Comment:
It is as easy as

1. copying  `salt/hosts/init.sls` to `salt/hosts-master/init.sls` and `salt/hosts-minion/init.sls`, 
2. modifying `top.sls`
```
base:
  '*':
  ...
  'roles:kube-master':
    - match: grain
    - hosts-minion
    ....
  'roles:kube-minion':
    - match: grain
    - hosts-minion
   ...
```

3. and then using this for `salt/hosts-minion/init.sls`:

```
+{%- for master_id, addrlist in salt['mine.get']('role:kube-master', 'network.ip_addrs').items() %}
+{{ master_id }}-host-entry:
   host.present:
 {% if addrlist is string %}
     - ip: {{ addrlist }}
 {% else %}
     - ip: {{ addrlist|first }}
 {% endif %}
    - names:
        - {{ master_id }}
        - api
        - api.{{ pillar['internal_infra_domain'] }}
{% endfor %}

```